CAMEL-24276: Azure OpenAI support for Camel JBang TUI AI prompt (#25171)
apache/camel: docs/user-manual/modules/ROOT/pages/camel-4x-upgrade-guide-4_22.adoc

The Camel JBang CLI now automatic resolves quarkus version to use,
instead of hardcoded `3.33.1.1` (https://github.com/apache/camel/commit/d1f4713ebdf787ab04a31c50a6f07e3ca66f0c2b).

The Camel CLI and TUI AI prompt (`camel ask`, TUI F8 panel) now auto-detect **Azure OpenAI**
when `AZURE_OPENAI_API_KEY` and `AZURE_OPENAI_ENDPOINT` are set (optional `AZURE_OPENAI_DEPLOYMENT_NAME`
and `AZURE_OPENAI_API_VERSION`). Azure requests use the `api-key` header instead of `Authorization: Bearer`.

**GitHub Models** is opt-in: set `GITHUB_MODELS=1` (or any non-false value) together with `GITHUB_TOKEN`

> to use `https://models.github.ai/inference`. A bare `GITHUB_TOKEN` alone no longer selects an LLM provider.

Reviewer: Nit: the wording "no longer selects" implies this was previous behavior that changed, but `GITHUB_TOKEN` was never used for LLM auto-detection in prior releases. Consider rephrasing to present tense:

```suggestion
to use `https://models.github.ai/inference`. A bare `GITHUB_TOKEN` alone does not select an LLM provider — the opt-in `GITHUB_MODELS` flag is required.
```
